code/datums/components/autofill.dm

```diff
@@ -0,0 +1,89 @@
+/// Automatically fills something with reagents... and possible leaks it
+TYPEINFO(/datum/component/auto_reagent)
+	initialization_args = list(
+		ARG_INFO("reagent_id", DATA_INPUT_TEXT, "Reagent to overflow/produce"),
+		ARG_INFO("units", DATA_INPUT_NUM, "Units to produce per cycle", 10),
+		ARG_INFO("overflowing", DATA_INPUT_BOOL, "Should overflow?", FALSE),
+	)
+/datum/component/auto_reagent
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// Reagent ID to produce
+	var/reagent_id = null
+	/// Number of reagents to produce per item process
+	var/units = 10
+	var/overflowing = FALSE
+	var/_added_to_items_processing = FALSE
+
+	Initialize(reagent_id=null, units=10, overflowing=FALSE)
+		if(!istype(parent,/atom))
+			return COMPONENT_INCOMPATIBLE
+		. = ..()
+		src.reagent_id = reagent_id
+		src.units = units
+		src.overflowing = overflowing
+
+		if(!istext(src.reagent_id) || isnull(reagents_cache[src.reagent_id]))
+			return
+		if(src.units <= 0)
+			return
```

Comment:
Shouldn't this return some failure? This way the component will still get created but registering of signals won't happen.

Reply:
Definitly should.  I convinced myself `COMPONENT_INCOMPATIBLE` was intended for incompatible types, and looking at how `Initialize()` is was off in the weeds.